### ISSUE TITLE
Authentication via passport.js

### ISSUE THREAD

**Issue by banglashi** [spacedeck/spacedeck-open#166]
Hi @mntmn 
how do you feel about using http://www.passportjs.org/ for authentication? If this is something you would be ok with I can start working on a pull request. I would replicate the username & password functionality first and later on build an ldap authentication.

**Comment by tekhnee** [comment]
There used to be (still are) placeholders in the code for Google login, but a generic OAuth solution would be terrific.

**Comment by Lukeva** [comment]
This would be awesome! 
@tekhnee Does Google Login work in the current version? I would love to test it, but can't really find the place where to set it up.

**Comment by yrammos** [comment]
It's been some time but, if memory serves, there were only indications that the feature may have been under consideration.

**Comment by tekhnee** [comment]
It's been some time but, if memory serves, there were only indications that the feature may have been under consideration.

**Comment by mntmn** [comment]
@banglashi yes, this is a good idea! I would like to be able to authenticate to Mattermost, for example. I would want to have this be configurable -- personally I wouldn't want to offer social media (facebook etc) auth on spacedeck.com.

**Comment by banglashi** [comment]
Hi @mntmn 
I started with it on this branch https://github.com/arillo/spacedeck-open/tree/feature/passport. First off I will reimplement the email/password authentication. Next on my list would be LDAP as a branch/module or just docs on how to implement. Mattermost would be OAuth, right? It should be easy to developt a passport plugin for it. Will keep you posted once I have a working version.

**Comment by rasos** [comment]
Generic oAuth would be awesome. We use keycloak as an IDM and let then other open source services. We would love to combine Spacedeck with Rocket.Chat or integrate into [BigBlueButton](https://git.fairkom.net/chat/fairblue/-/issues/72#note_31089).

**Comment by r9labs** [comment]
@banglashi did you get it working? I think this is almost a critical feature.

**Comment by banglashi** [comment]
Hi @r9labs 
It's been a while... I got it working on this branch https://github.com/arillo/spacedeck-open/tree/feature/passport but there is still the old session mechanism working since it was used in many places and I didn't have the time to properly refactor it. The branch would also need to be updated to the latest changes. I can try to take a look at the end of the month, currently finishing up some other projects. Feel free to take a look.
